partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelExclude.kt

```diff
@@ -0,0 +1,184 @@
+package org.partiql.eval.internal.operator.rel
+
+import org.partiql.eval.internal.Record
+import org.partiql.eval.internal.exclude.CompiledExcludeItem
+import org.partiql.eval.internal.exclude.ExcludeFieldCase
+import org.partiql.eval.internal.exclude.ExcludeNode
+import org.partiql.eval.internal.exclude.ExcludeStep
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.value.BagValue
+import org.partiql.value.CollectionValue
+import org.partiql.value.ListValue
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.SexpValue
+import org.partiql.value.StructValue
+import org.partiql.value.bagValue
+import org.partiql.value.listValue
+import org.partiql.value.sexpValue
+import org.partiql.value.structValue
+
+internal class RelExclude(
+    private val input: Operator.Relation,
+    private val exclusions: List<CompiledExcludeItem>
+) : Operator.Relation {
+
+    override fun open() {
+        input.open()
+    }
+
+    override fun next(): Record? {
+        while (true) {
+            val row = input.next() ?: return null
+            val newRecord = exclusions.fold(row) { curRecord, expr ->
+                excludeOnRecord(curRecord, expr)
+            }
+            return newRecord
+        }
+    }
+
+    override fun close() {
+        input.close()
+    }
+
+    @OptIn(PartiQLValueExperimental::class)
+    private fun excludeOnRecord(
+        record: Record,
+        exclusions: CompiledExcludeItem
+    ): Record {
+        val values = record.values
+        val value = values.getOrNull(exclusions.root)
+        val newValues = if (value != null) {
+            values[exclusions.root] = excludeOnPartiQLValue(value, exclusions)
+            values
+        } else {
+            values
+        }
+        return Record(newValues)
+    }
+
+    @OptIn(PartiQLValueExperimental::class)
+    private fun excludeOnStructValue(
+        structValue: StructValue<*>,
+        exclusions: ExcludeNode
+    ): PartiQLValue {
+        val attrsToRemove = exclusions.leaves.mapNotNull { leaf ->
+            when (val leafStep = leaf.step) {
+                is ExcludeStep.StructWildcard -> {
+                    // tuple wildcard at current level. return empty struct
+                    return structValue<PartiQLValue>()
+                }
+                is ExcludeStep.StructField -> leafStep.attr
+                is ExcludeStep.CollIndex, is ExcludeStep.CollWildcard -> null
+            }
+        }.toSet()
+        val branches = exclusions.branches
+        val finalStruct = structValue.entries.mapNotNull { structField ->
+            if (attrsToRemove.contains(structField.first)) {
+                // struct attr is to be removed at current level
+                null
+            } else {
+                // deeper level exclusions
+                val name = structField.first
+                var expr = structField.second
```

Comment:
The second is not an expression, it's a value. 


```suggestion
                val (key, value) = structField
```